src/test/java/cloud/fogbow/common/plugins/cloudidp/cloudstack/CloudStackIdentityProviderPluginTest.java

```diff
@@ -0,0 +1,289 @@
+	// without a user name, it must throw an InvalidParameterException.
+	@Test(expected = InvalidParameterException.class) // verify
+	public void testGetCloudUserWithoutUsername() throws FogbowException {
+		// set up
+		Map<String, String> credentials = new HashMap<String, String>();
+		credentials.put(CloudStackConstants.Identity.PASSWORD_KEY_JSON, ANY_VALUE);
+		credentials.put(CloudStackConstants.Identity.DOMAIN_KEY_JSON, ANY_VALUE);
+
+		// exercise
+		this.plugin.getCloudUser(credentials);
+	}
+
+	// test case: When invoking the getCloudUser method with a credential map
+	// without a password, it must throw an InvalidParameterException.
+	@Test(expected = InvalidParameterException.class) // verify
+	public void testGetCloudUserWithoutPassword() throws FogbowException {
+		// set up
+		Map<String, String> credentials = new HashMap<String, String>();
+		credentials.put(CloudStackConstants.Identity.USERNAME_KEY_JSON, ANY_VALUE);
+		credentials.put(CloudStackConstants.Identity.DOMAIN_KEY_JSON, ANY_VALUE);
+
+		// exercise
+		this.plugin.getCloudUser(credentials);
+	}
+
+	// test case: When invoking the getCloudUser method with a credential map
+	// without a domain, it must throw an InvalidParameterException.
+	@Test(expected = InvalidParameterException.class) // verify
+	public void testGetCloudUserWithoutDomain() throws FogbowException {
+		// set up
+		Map<String, String> credentials = new HashMap<String, String>();
+		credentials.put(CloudStackConstants.Identity.USERNAME_KEY_JSON, ANY_VALUE);
+		credentials.put(CloudStackConstants.Identity.PASSWORD_KEY_JSON, ANY_VALUE);
+
+		// exercise
+		this.plugin.getCloudUser(credentials);
+	}
+
+	// test case: When invoking the getCloudUser method with a valid credential map,
+	// and the response to the login request in the server returns an HTTP error
+	// code, it must throw a FogbowException.
+	@Test(expected = FogbowException.class)
+	public void testGetCloudUserWithHttpCodeErrorFromLoginResponse() throws FogbowException {
+		// set up
+		Map<String, String> credentials = new HashMap<String, String>();
+		credentials.put(CloudStackConstants.Identity.USERNAME_KEY_JSON, ANY_VALUE);
+		credentials.put(CloudStackConstants.Identity.PASSWORD_KEY_JSON, ANY_VALUE);
+		credentials.put(CloudStackConstants.Identity.DOMAIN_KEY_JSON, ANY_VALUE);
+
+		int httpCode = HTTP_CODE_ERROR;
+		HttpResponse response = createLoginResponse(httpCode);
+		Mockito.doReturn(response).when(this.plugin).doLoginAuthentication(Mockito.any(LoginRequest.class));
+
+		// exercise
+		this.plugin.getCloudUser(credentials);
+	}
+	
+	// test case: When invoking the getCloudUser method with a valid credential map,
+	// and the response returns an HTTP error code from an accounts list request in
+	// the server, it must throw a FogbowException.
+	@Test(expected = FogbowException.class) // verify
+	public void testGetCloudUserWithHttpCodeErrorFromListAccountsResponse() throws FogbowException {
+		// set up
```

Comment:
same problem of the above test case. If we testing the case of errors in the createLoginRequest method, there is no need to mock the behaviour of its inner method. Mock the own createLoginRequest method